apps/pwabuilder/src/script/components/share-card.ts

```diff
@@ -0,0 +1,473 @@
+        </div>      `
+      return;
+    }
+    if(/Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent) ) {
+      //console.log("This is a mobile browser with the Share API available.");
+      //this.shadowRoot!.getElementById("share-actions")!.style.flexDirection = 'column-reverse';
+      this.actionButtons = html`
+        <div>
+          <button type="button" id="copy-button" class="standard-button secondary" @click=${() => this.copyImage()}><img class="actions-icons" src="/assets/copy-icon-standard-color.svg" alt="Download image button icon"/></button>
+          <span id="copy-button-label" class="standard-button-label">Copy</span>
+        </div>
+        <div>
+          <button type="button" id="download-button" class="standard-button secondary" @click=${() => this.htmlToImage('download')}><img class="actions-icons" src="/assets/download-icon-standard-color.svg" alt="Download image button icon"/></button>
+          <span id="download-button-label" class="standard-button-label">Download</span>
+        </div>        
+        <div>
+          <button type="button" id="share-button" class="standard-button secondary" @click=${() => this.htmlToImage('share')}><img class="actions-icons" src="/assets/share_icon.svg" alt="Share image button icon"/></button>  
+          <span class="standard-button-label">Share</span>
+        </div>           `
+    } else {
+        //console.log("This is a desktop browser with the Share API available.");
+        this.actionButtons = html`
+        <div>
+          <button type="button" id="copy-button" class="standard-button secondary" @click=${() => this.copyImage()}><img class="actions-icons" src="/assets/copy-icon-standard-color.svg" alt="Download image button icon"/></button>
+          <span id="copy-button-label" class="standard-button-label">Copy</span>
+        </div>
+        <div>
+          <button type="button" id="download-button" class="standard-button secondary" @click=${() => this.htmlToImage('download')}><img class="actions-icons" src="/assets/download-icon-standard-color.svg" alt="Download image button icon"/></button>
+          <span id="download-button-label" class="standard-button-label">Download</span>
+        </div>
+        <div>
+          <button type="button" id="share-button" class="standard-button secondary" @click=${() => this.htmlToImage('share')}><img class="actions-icons" src="/assets/share_icon.svg" alt="Share image button icon"/></button>  
+          <span class="standard-button-label">Share</span>
+        </div>                          
+      `
+    }
+    return;
+  }
+
+  render() {
+    return html`
+      <sl-dialog class="dialog" @sl-show=${() => this.setup()} @sl-hide=${() => this.hideDialog()} noHeader>
+        <div class="share-modal-header">Share this image with the community!</div>
+        <div id="frame-wrapper">
+          <div id="frame-content">
+            <div id="canvas-holder">
+              <canvas
```

Comment:
Try to change this.shadowRoot!.getElementById("myCanvas")
to this mechanic https://lit.dev/docs/templates/directives/#referencing-rendered-dom
<canvas ${ref(this.shareCanvas)}>

Reply:
Ref added